### PR TITLE
Referral routing

### PR DESCRIPTION
**Pull request by stephencelis**
This updates the user model and adds routing for referrals. This does _not_ introduce any logic around referrals yet.

### CODE REVIEW

File: Sources/Database/Database.swift [hunk 1]
```diff
@@ -363,15 +363,7 @@ private struct _Client {
...
-    SELECT "users"."email",
-           "users"."episode_credit_count",
-           "users"."github_user_id",
-           "users"."github_access_token",
-           "users"."id",
-           "users"."is_admin",
-           "users"."name",
-           "users"."subscription_id",
-           "users"."rss_salt"
+    SELECT "users".*
```
Comment (stephencelis) [inline Sources/Database/Database.swift]: Database decoding errors are super opaque and we get them whenever we add new fields to the user model so I think `*` is prob the way to go for these queries...

File: Sources/Models/SubscribeConfirmationData.swift [hunk 2]
```diff
@@ -4,15 +4,18 @@ import Stripe
 public struct SubscribeConfirmationData: Equatable {
   public var billing: Pricing.Billing
   public var isOwnerTakingSeat: Bool
+  public var referralCode: User.ReferralCode?
```
Comment (stephencelis) [inline Sources/Models/SubscribeConfirmationData.swift]: Referral code now comes along for the ride here...

File: Sources/PointFreeRouter/Routes.swift [hunk 3]
```diff
@@ -308,20 +310,23 @@ private let subscriberDataIso = PartialIso<String, SubscribeData?>(
...
     let isOwnerTakingSeat = keyValues
-      .first { key, value in key == SubscribeData.CodingKeys.isOwnerTakingSeat.rawValue }?.1
+      .first { k, _ in k == SubscribeData.CodingKeys.isOwnerTakingSeat.rawValue }?.1
       .flatMap(Bool.init)
       ?? false
 
     let rawCouponValue = keyValues.first(where: { key, value in key == "coupon" })?.1
     let coupon = rawCouponValue == "" ? nil : rawCouponValue.flatMap(Coupon.Id.init(rawValue:))
+    let referralCode = keyValues
+      .first(where: { key, value in key == SubscribeData.CodingKeys.referralCode.rawValue })?.1
```
Comment (mbrandonw) [inline Sources/PointFreeRouter/Routes.swift]: I guess this value is also not needed right?

```suggestion
      .first(where: { key, _ in key == SubscribeData.CodingKeys.referralCode.rawValue })?.1
```

File: Tests/PointFreeTests/SubscribeConfirmationTests.swift [hunk 4]
```diff
@@ -355,7 +364,8 @@ class SubscriptionConfirmationTests: TestCase {
           lane: .team,
           billing: nil,
           isOwnerTakingSeat: nil,
-          teammates: nil
+          teammates: nil,
+          referralCode: nil
```
Comment (mbrandonw) [inline Tests/PointFreeTests/SubscribeConfirmationTests.swift]: is it possible to make a test for when the referral code is present?

Comment (stephencelis) [reply Tests/PointFreeTests/SubscribeConfirmationTests.swift]: I'm gonna add tests when the logic is actually in place. Not sure what is possible to test at this point...

Comment (mbrandonw) [reply Tests/PointFreeTests/SubscribeConfirmationTests.swift]: it's testing the gnarly logic in the `subscriberDataIso` for the route: 

https://github.com/pointfreeco/pointfreeco/pull/570/files#diff-549405fc9f778f602011a17f333a61c5R308

that stuff is still just a hacky workaround to our lack of properly decoding that info from the URL.

**Pull request merged**